test/module/irohad/synchronizer/synchronizer_test.cpp

```diff
@@ -110,14 +116,87 @@ class SynchronizerTest : public ::testing::Test {
   std::shared_ptr<shared_model::interface::Block> commit_message;
   shared_model::interface::types::PublicKeyCollectionType public_keys;
   shared_model::interface::types::HashType hash;
-  std::shared_ptr<shared_model::interface::Peer> peer;
   std::shared_ptr<PeerList> ledger_peers;
+  std::vector<shared_model::crypto::Keypair> ledger_peer_keys;
 
   rxcpp::subjects::subject<ConsensusGate::GateObject> gate_outcome;
 
   std::shared_ptr<SynchronizerImpl> synchronizer;
 };
 
+class ChainMatcher : public ::testing::MatcherInterface<Chain> {
+ public:
+  explicit ChainMatcher(
+      std::vector<std::shared_ptr<shared_model::interface::Block>>
+          expected_chain)
+      : expected_chain_(std::move(expected_chain)) {}
+
+  bool MatchAndExplain(
+      Chain test_chain,
+      ::testing::MatchResultListener *listener) const override {
+    size_t got_blocks = 0;
+    auto wrapper =
+        make_test_subscriber<CallExact>(test_chain, expected_chain_.size());
+    wrapper.subscribe([this, &got_blocks](const auto &block) {
+      EXPECT_LT(got_blocks, expected_chain_.size())
+          << "Tested chain provides more blocks than expected";
+      if (got_blocks < expected_chain_.size()) {
+        const auto &expected_block = expected_chain_[got_blocks];
+        EXPECT_THAT(*block, Eq(ByRef(*expected_block)))
+            << "Block number " << got_blocks << " does not match!";
+      }
+      ++got_blocks;
+    });
+    return wrapper.validate();
+  }
+
+  virtual void DescribeTo(::std::ostream *os) const {
+    *os << "Tested chain matches expected chain.";
+  }
+
+  virtual void DescribeNegationTo(::std::ostream *os) const {
+    *os << "Tested chain does not match expected chain.";
+  }
+
+ private:
+  const std::vector<std::shared_ptr<shared_model::interface::Block>>
+      expected_chain_;
+};
+
+inline ::testing::Matcher<Chain> ChainEq(
+    std::vector<std::shared_ptr<shared_model::interface::Block>>
+        expected_chain) {
+  return ::testing::MakeMatcher(new ChainMatcher(expected_chain));
+}
+
+void mutableStorageExpectChain(
+    iroha::ametsuchi::MockMutableFactory &mutable_factory,
+    std::vector<std::shared_ptr<shared_model::interface::Block>> chain) {
+  const bool must_create_storage = not chain.empty();
```

Comment:
`must_create_storage` sounds misleading, because the code below uses `WillRepeatedly` if `must_create_storage` is false. Should it be `must_create_once`?